### PR TITLE
Add support for `lib.exe`

### PR DESCRIPTION
**Pull request by messense**
https://docs.microsoft.com/en-us/cpp/build/reference/building-an-import-library-and-export-file?view=msvc-170

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,3 +9,6 @@ license = "MIT AND PSF-2.0"
 keywords = ["build-dependencies", "python", "windows", "mingw"]
 categories = ["development-tools::build-utils"]
 readme = "README.md"
+
+[dependencies]
```
Comment (ravenexp) [inline Cargo.toml]: Maybe it's better to make this dep conditional as:

`[target.'cfg(windows)'.dependencies]`

File: src/lib.rs [hunk 2]
```diff
@@ -155,12 +164,28 @@ pub fn generate_implib_for_target(out_dir: &Path, arch: &str, env: &str) -> Resu
     }
 }
 
+/// Find Visual Studio lib.exe
+fn find_lib_exe(arch: &str) -> Option<Command> {
```
Comment (ravenexp) [inline src/lib.rs:169]: This function can be made `#[cfg(windows)]`, and you can add a stub like
```rust
`#[cfg(not(windows))]`
fn find_lib_exe(_arch: &str) -> Option<Command> {
    None
}
```

File: src/lib.rs [hunk 3]
```diff
@@ -155,12 +164,28 @@ pub fn generate_implib_for_target(out_dir: &Path, arch: &str, env: &str) -> Resu
...
+        "x86" => "i686-pc-windows-msvc",
+        "aarch64" => "aarch64-pc-windows-msvc",
+        _ => return None,
+    };
+    cc::windows_registry::find(target, LIB_MSVC)
+}
+
 /// Generates the complete `dlltool` executable invocation command.
 ///
 /// Supports both LLVM and MinGW `dlltool` flavors.
```
Comment (ravenexp) [inline src/lib.rs]: You should also mention MSVC's `lib.exe` in the comment.

**Review by ravenexp — CHANGES_REQUESTED** [review]
Looks good to me overall. I suggested using conditional compilation to reduce footprint for non-windows targets.

Also, I've been in the process of adding some basic CI testing (#1), so it probably makes sense to add a new CI job that runs the crate unit tests in the `windows-latest` environment as well. It has [MSVC 17.1 pre-installed](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2022-Readme.md#visual-studio-enterprise-2022), so everything should hopefully work out of the box.

**Review by ravenexp — APPROVED** [review]
Nice, thanks!

Edit: MinGW dlltool is not found on `windows-latest`, which leads to partial unit test failures. Maybe these tests can be made conditional on `#[cfg(unix)]` or you could figure out how to install MinGW binutils via MSYS2 `pacman`.

**Comment by messense** [comment]
Since mingw64 is already tested on Linux, I choose to add `#[cfg(unix)]`s.

**Review by ravenexp — APPROVED** [review]

**Comment by ravenexp** [comment]
Everything seems fine, thank you!

I'll release it as `0.2.1` in a few days, after I figure out how to publish crates from GitHub Actions.

**Comment by messense** [comment]
> after I figure out how to publish crates from GitHub Actions.

Here is an example: https://github.com/PyO3/maturin/blob/e289772a137a07a6feea3c1c53dd99db6e800830/.github/workflows/release.yml#L8-L33

**Comment by ravenexp** [comment]
> Here is an example: https://github.com/PyO3/maturin/blob/e289772a137a07a6feea3c1c53dd99db6e800830/.github/workflows/release.yml#L8-L33

Thanks! Just published `0.2.1` with https://github.com/ravenexp/python3-dll-a/runs/6054441360?check_suite_focus=true

**Pull request merged**